File: core/src/main/java/io/github/krys/nextrtp/core/command/ReloadCommand.java
package io.github.krys.nextrtp.core.command;

import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import com.mojang.brigadier.context.CommandContext;
import com.mojang.brigadier.tree.CommandNode;
import io.github.krys.nextrtp.common.command.BrigadierCommand;
import io.github.krys.nextrtp.core.ReloadableManager;
import io.github.krys.nextrtp.core.message.Messager;
import io.papermc.paper.command.brigadier.CommandSourceStack;
import org.jetbrains.annotations.NotNull;

public final class ReloadCommand implements BrigadierCommand {

  private final @NotNull ReloadableManager manager;

  public ReloadCommand(@NotNull ReloadableManager manager) {
    this.manager = manager;
  }

  @Override
  public CommandNode<CommandSourceStack> node() {
    LiteralArgumentBuilder<CommandSourceStack> builder = literal("reload")
            .requires(source -> source.getSender().hasPermission("nextrtp.command.reload"))
            .executes(this::execute);

    return builder.build();
  }

  private int execute(CommandContext<CommandSourceStack> context) {
    try {
      manager.reload();
      Messager.RELOAD_SUCCESS.accept(context.getSource().getSender());
    } catch (Exception e) {
      Messager.RELOAD_ERROR.accept(context.getSource().getSender());
      throw new RuntimeException(e);
    }
    return 1;
  }
}

File: src/main/kotlin/ur/kryz/rtp/command/subcommand/ReloadCommand.kt
package ur.kryz.rtp.command.subcommand

import com.mojang.brigadier.Command
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import com.mojang.brigadier.context.CommandContext
import io.papermc.paper.command.brigadier.CommandSourceStack
import io.papermc.paper.command.brigadier.Commands
import ur.kryz.rtp.NextRTPPlugin
import ur.kryz.rtp.command.paper.SubCommand
import ur.kryz.rtp.utils.Permissions
import ur.kryz.rtp.utils.PlayerUtil

class ReloadCommand(
    private val plugin: NextRTPPlugin
) : SubCommand<CommandSourceStack> {

    override fun execute(context: CommandContext<CommandSourceStack>) : Int {
        val sender = context.source.sender
        plugin.reload()
        PlayerUtil.sendParsed(sender,"messages.command.reload.success")
        return Command.SINGLE_SUCCESS
    }

    override fun build(builder: LiteralArgumentBuilder<CommandSourceStack>) : LiteralArgumentBuilder<CommandSourceStack> {
        return builder.then(
            Commands.literal("reload")
                .requires { req ->
                    Permissions.anyPermissions(
                        req,"nextrtp.command.reload"
                    )
                }
                .executes(this::execute)
        )
    }
}